ipatests/test_integration/test_sssd.py

```diff
@@ -403,3 +405,186 @@ def test_trustdomain_disable_disables_subdomain(self):
+            pytest.fail('Authentication using SSH key not successful', e)
+
+        # add member
+        cmd_output = master.run_command(['ipa', 'group-add-member',
+                                         'b', '--users', username])
+        assert 'Group name: b' in cmd_output.stdout_text
+        assert 'Member users: {}'.format(username) in cmd_output.stdout_text
+        assert 'Member groups: a' in cmd_output.stdout_text
+        assert 'Member of groups: c' in cmd_output.stdout_text
+        assert 'Number of members added 1' in cmd_output.stdout_text
+
+        # now check ssh on the client
+        clear_sssd_cache(client)
+        client.put_file_contents('/tmp/user_ssh_priv_key',
+                                 ssh_key_file_contents)
+        client.run_command(['chmod', '600', '/tmp/user_ssh_priv_key'])
+
+        # after adding testuser001 to b group
+        # testuser001 will have the following memberships
+        # Member of groups: a, ipausers, b
+        # Indirect Member of group: c
+        try:
+            cmd = ['ssh', '-i', '/tmp/user_ssh_priv_key',
+                   '-q', '{}@{}'.format(username, client.hostname), 'groups']
+            cmd_output = client.run_command(cmd)
+            assert '{}'.format(username) in cmd_output.stdout_text
+            assert 'a' in cmd_output.stdout_text
+            assert 'b' in cmd_output.stdout_text
+            assert 'c' in cmd_output.stdout_text
+        except Exception as e:
+            pytest.fail('Authentication using SSH key not successful', e)
+
+        # now back to server to remove member
+        cmd_output = master.run_command(['ipa', 'group-remove-member',
+                                         'b', '--users', username])
+        assert_str = 'Indirect Member users: {}'.format(username)
+        assert 'Group name: b' in cmd_output.stdout_text
+        assert 'Member groups: a' in cmd_output.stdout_text
+        assert 'Member of groups: c' in cmd_output.stdout_text
+        assert assert_str in cmd_output.stdout_text
+        assert 'Number of members removed 1' in cmd_output.stdout_text
+
+        clear_sssd_cache(master)
+
+        # now check ssh on the client again
+        # after removing testuser001 from b group
+        # testuser001 will have the following memberships
+        # Member of groups: a, ipausers
+        # Indirect Member of group: b, c
+        clear_sssd_cache(client)
+        try:
+            cmd = ['ssh', '-i', '/tmp/user_ssh_priv_key',
+                   '-q', '{}@{}'.format(username, client.hostname), 'groups']
+            cmd_output = client.run_command(cmd)
+            assert '{}'.format(username) in cmd_output.stdout_text
+            assert 'a' in cmd_output.stdout_text
+            assert 'b' in cmd_output.stdout_text
+            assert 'c' in cmd_output.stdout_text
+        except Exception as e:
+            pytest.fail('Authentication using SSH key not successful', e)
+
+        # test cleanup
```

Comment:
The cleanup should be executed even if the test fails. You can use a fixture or try/except.

Reply:
added cleanup as a fixture